modules/__tests__/TestSequences/EncodedReservedCharacters.js

```diff
@@ -0,0 +1,46 @@
+import expect from 'expect'
+import execSteps from './execSteps'
+
+export default (history, done) => {
+  const steps = [
+    () => {
+      // encoded string
+      const pathname = '/view/%23abc'
+      history.replace(pathname)
+    },
+    (location) => {
+      expect(location).toMatch({
+        pathname: '/view/%23abc'
+      })
+
+      // encoded object
+      const pathname = '/view/%23abc'
+      history.replace({ pathname })
+    },
+    (location) => {
+      expect(location).toMatch({
+        pathname: '/view/%23abc'
+      })
+      // unencoded string
+      const pathname = '/view/#abc'
+      history.replace(pathname)
+    }
+    ,
+    (location) => {
+      expect(location).toMatch({
+        pathname: '/view/',
+        hash: '#abc'
+      })
+      // unencoded object
+      const pathname = '/view/#abc'
+      history.replace({ pathname })
+    },
+    (location) => {
+      expect(location).toMatch({
+        pathname: '/view/#abc'
```

Comment:
Seems like the kind of thing we don't actually need to test. i.e. if anyone actually tried to do this I'd probably just tell them to use `hash` instead of putting it on the `pathname`.